sled-agent/src/services_migration.rs

```diff
@@ -1,623 +0,0 @@
-            self.service_address,
-        );
-
-        Ok(OmicronZoneDataset { pool_name: self.name.pool().clone() })
-    }
-}
-
-#[cfg(test)]
-mod test {
-    use super::AllZoneRequests;
-    use crate::services::OmicronZonesConfigLocal;
-    use camino::Utf8PathBuf;
-
-    /// Verifies that our understanding of this old-format ledger has not
-    /// changed.  (If you need to change this for some reason, you must figure
-    /// out how that affects systems with old-format ledgers and update this
-    /// test accordingly.)
-    #[test]
-    fn test_all_services_requests_schema() {
-        let schema = schemars::schema_for!(AllZoneRequests);
-        expectorate::assert_contents(
-            "../schema/all-zone-requests.json",
-            &serde_json::to_string_pretty(&schema).unwrap(),
-        );
-    }
-
-    /// Verifies that we can successfully convert a corpus of known old-format
-    /// ledgers.  These came from two racks operated by Oxide.  In practice
-    /// there probably aren't many different configurations represented here but
-    /// it's easy enough to just check them all.
-    ///
-    /// In terms of verifying the output: all we have done by hand in
-    /// constructing this test is verify that the code successfully converts
-    /// them.  The conversion code does some basic sanity checks as well, like
-    /// that we produced the same number of zones and datasets.
-    #[test]
-    fn test_convert_known_ledgers() {
-        let known_ledgers = &[
-            /* rack2 */
-            "rack2-sled8.json",
-            "rack2-sled9.json",
-            "rack2-sled10.json",
-            "rack2-sled11.json",
-            "rack2-sled12.json",
-            "rack2-sled14.json",
-            "rack2-sled16.json",
-            "rack2-sled17.json",
-            "rack2-sled21.json",
-            "rack2-sled23.json",
-            "rack2-sled25.json",
-            /* rack3 (no sled 10) */
-            "rack3-sled0.json",
-            "rack3-sled1.json",
-            "rack3-sled2.json",
-            "rack3-sled3.json",
-            "rack3-sled4.json",
-            "rack3-sled5.json",
-            "rack3-sled6.json",
-            "rack3-sled7.json",
-            "rack3-sled8.json",
-            "rack3-sled9.json",
-            "rack3-sled11.json",
-            "rack3-sled12.json",
-            "rack3-sled13.json",
-            "rack3-sled14.json",
-            "rack3-sled15.json",
-            "rack3-sled16.json",
-            "rack3-sled17.json",
-            "rack3-sled18.json",
-            "rack3-sled19.json",
-            "rack3-sled20.json",
-            "rack3-sled21.json",
-            "rack3-sled22.json",
-            "rack3-sled23.json",
-            "rack3-sled24.json",
-            "rack3-sled25.json",
-            "rack3-sled26.json",
-            "rack3-sled27.json",
-            "rack3-sled28.json",
-            "rack3-sled29.json",
-            "rack3-sled30.json",
-            "rack3-sled31.json",
```

Comment:
I think we can probably remove all of these files, too -- basically all of `omicron/sled-agent/tests/old-service-ledgers` and `omicron/sled-agent/tests/output/new-service-ledgers`.

Reply:
Good catch, definitely should have removed these -- done!